### PR TITLE
[FIX] project: fix sol creation from task

### PR DESCRIPTION
**Pull request by rish-odoo**
Steps:
    - Install project & sales app
    - Open project app
    - Open Project
    - Select any project
    - Add sale order item

    Issue:
    - - the creation of SOLs on the fly , doesn't seem to be working anymore

    Cause:
    - There is no option for creating new sale order item

    Fix:
    - Appear create and edit the option for creation of Sale order item

   Task-3470877

### CODE REVIEW

File: addons/sale_project/views/project_task_views.xml [hunk 1]
```diff
@@ -213,7 +213,7 @@
                 <field name="sale_line_id"
                     groups="sales_team.group_sale_salesman"
                     string="Sales Order Item"
-                    options='{"no_create": True}'
+                    options='{"no_quick_create": True}'
```
Comment (MissingNoShiny) [inline addons/sale_project/views/project_task_views.xml:216]:
```suggestion
```
Same comment here

File: addons/sale_project/models/sale_order_line.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ def default_get(self, fields):
                 except AccessError:
                     del res['order_id']
 
-            if 'order_id' in fields and 'order_id' not in res:
+            if 'order_id' in fields and ('order_id' not in res or not res.get('order_id')):
```
Comment (MissingNoShiny) [inline addons/sale_project/models/sale_order_line.py:40]: What's the point of this change?

File: addons/sale_project/models/sale_order_line.py [hunk 3]
```diff
@@ -60,7 +60,7 @@ def default_get(self, fields):
                     }
 
                 if not sale_order:
-                    company_id = self.env.context.get('defaut_company_id', False)
+                    company_id = self.env.context.get('defaut_company_id', self.env.company.id)
```
Comment (MissingNoShiny) [inline addons/sale_project/models/sale_order_line.py:63]: I think you can leave the `False`, there is however a typo in the context key :sweat_smile: 
```suggestion
                    company_id = self.env.context.get('default_company_id', False)
```

**Review by MissingNoShiny — COMMENTED** [review]
Thanks for the fix, I left a few comments :slightly_smiling_face:

File: addons/sale_project/models/sale_order_line.py (hunk 2 above)
Comment (rish-odoo) [reply addons/sale_project/models/sale_order_line.py:40]: This whole condition check if order_id not in fields or present in fields and if either the key 'order_id' is missing in the res dictionary or the value of 'order_id' in res is false

Comment (MissingNoShiny) [reply addons/sale_project/models/sale_order_line.py:40]: I did some testing and I now see why it is necessary. However, since we want to check if the value is False, the condition is a bit redundant, and you could just keep this:
```suggestion
            if 'order_id' in fields and not res.get('order_id'):
```

File: addons/sale_project/models/sale_order_line.py (hunk 3 above)
Comment (MissingNoShiny) [reply addons/sale_project/models/sale_order_line.py:63]: I don't know why it wouldn't work, strange :confused: 
I will have a look tomorrow

Comment (MissingNoShiny) [reply addons/sale_project/models/sale_order_line.py:63]: After investigating, it turns out that it keeps the `default_company_id` context key in the create, which has a value of `False` when the project has no company. Therefore, we will indeed need to keep the default value you added. The code can however be simplified a bit, because then we know that we will always provide the create with a company_id. I would suggest something like this:
```py
            if 'order_id' in fields and not res.get('order_id'):
                assert (partner_id := self.env.context.get('default_partner_id'))
                project_id = self.env.context.get('link_to_project')
                sale_order = None
                so_create_values = {
                    'partner_id': partner_id,
                    'company_id': company_id = self.env.context.get('defaut_company_id', self.env.company.id),
                }
                if project_id:
                    try:
                        project_so = self.env['project.project'].browse(project_id).sale_order_id
                        project_so.check_access_rule('write')
                        sale_order = project_so
                    except AccessError:
                        pass

                    if not sale_order:
                        so_create_values['project_ids'] = [Command.link(project_id)]

                if not sale_order:
                    sale_order = self.env['sale.order'].create(so_create_values)
                    sale_order.action_confirm()
                default_values['order_id'] = sale_order.id
```
(I didn't test this code)

Comment (MissingNoShiny) [reply addons/sale_project/models/sale_order_line.py:63]: (Also, don't forget to fix the typo in the context key)

File: addons/sale_project/models/sale_order_line.py [hunk 4]
```diff
@@ -37,35 +37,28 @@ def default_get(self, fields):
...
-
+                        so_create_values['project_ids'] = [Command.link(project_id)]
                 if not sale_order:
-                    company_id = self.env.context.get('defaut_company_id', False)
-                    if company_id:
-                        so_create_values['company_id'] = company_id
                     sale_order = self.env['sale.order'].create(so_create_values)
                     sale_order.action_confirm()
                 default_values['order_id'] = sale_order.id
+
```
Comment (MissingNoShiny) [inline addons/sale_project/models/sale_order_line.py:61]: Unnecessary change :innocent:

**Review by MissingNoShiny — COMMENTED** [review]
LGTMn thanks for your work. :slightly_smiling_face: 
I just have minor formatting suggestions

**Review by xavierbol — APPROVED** [review]
LGTM, thanks for your work. 🙂 

robodoo r+
